dc/s2n-quic-dc/src/path/secret/map/handshake.rs

```diff
@@ -84,7 +84,14 @@ impl dc::Path for HandshakingPath {
         &mut self,
         session: &impl s2n_quic_core::crypto::tls::TlsSession,
     ) -> Result<Vec<s2n_quic_core::stateless_reset::Token>, s2n_quic_core::transport::Error> {
-        self.application_data = Some(self.map.store.application_data(session));
+        match self.map.store.application_data(session) {
+            Ok(application_data) => {
+                self.application_data = application_data;
```

Comment:
You need to update the Entry (dc/s2n-quic-dc/src/path/secret/map/entry.rs) to also have a `Option<ApplicationData>` -- otherwise this will panic below on a take().unwrap() of this field.